src/gui/qt.hpp

```diff
@@ -0,0 +1,27 @@
+#ifndef CALIBRATOR_WORKER_H
+#define CALIBRATOR_WORKER_H
+
+#include <QObject>
+#include <QScopedPointer>
+
+class Calibrator;
+class CalibratorWorker : public QObject
+{
+    Q_OBJECT
+public:
+    CalibratorWorker(int argc, char **argv, QObject *parent = nullptr);
+    ~CalibratorWorker();
+
+    Q_INVOKABLE void init();
+
+    Q_INVOKABLE void onClicked(int x, int y, int displayWidth, int displayHeight);
+
+signals:
+    void stateChanged(int state);
+    void setDisplaySize(int newWidth, int newHeight, int numBlocks);
```

Comment:
setDisplaySize is not the best name for signal (due to handler naming onSetDisplaySize for example and overall idea of notifying about something by signals). Something like displaySizeShouldBeUpdated/initDone/initialized?